Read GHRSST Level 4 NetCDF data files

twsearle opened the pull request:
This is some work I did as part of the hackathon adding some basic support for reading sea surface temperature analysis data from the [GHRSST project](www.ghrsst.org). The main changes are in `ghrsstl4.py` and I have added some tests in `test_ghrsstl4.py` (pretty much copied from `gridded_forecast.py` and `test_gridded_forecast.py`).

I have altered `geo.py` so that any mask on the data is preserved in `stretch_image`.

To test, I have been running:
`forest --show --file-type ghrsstl4 20190918120000-UKMO-L4_GHRSST-SSTfnd-OSTIA-GLOB-v02.0-fv02.0.nc`

OSTIA data ought to conform to the [GHRSST data specification](https://www.ghrsst.org/about-ghrsst/governance-documents/), but there a few other level 4 analyses available from other met centres which also seem to plot OK (GAMSSA and OSPO_night are two others I checked).

### Issues
  * These are not forecasts, so I set forecast time to 1970-01-01, but this is a bit ugly
  * The colour bars start from the missing data indicator values rather than the beginning of the real data. I haven't been able to work out what is doing this yet.
  * Many, many others I didn't notice :)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/informatics-lab/forest?pullRequest=185) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/informatics-lab/forest?pullRequest=185) before we can accept your contribution.<br/><hr/>**toby.searle** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/informatics-lab/forest?pullRequest=185) it.</sub>

andrewgryan commented:
@twsearle this looks great! I think the only thing missing is your git command line account settings and github account aren't lined up, this happened for @caughtshort in #183 too

- `git config --global user.name twsearle`
- `git config --global user.email ...@metoffice.gov.uk`

As for amending history, `git rebase -i` and `git commit --ammend --author "Author Name<author.email@hello.com>"` can be used. Or maybe, `git checkout a12c3d` followed by `git commit --amend`, I've not done it before so have pasted a SO comment thread. Naturally, if you are n't concerned about your commit statistics then don't worry about editing the commit history

https://stackoverflow.com/questions/3042437/how-to-change-the-commit-author-for-one-specific-commit

twsearle commented:
@andrewgryan yes, sorry about that! I am having a nightmare trying to get my git configuration to be compatible with my old github account and the ECMWF NEMOVAR repository account name and keys.

I think I have fixed things now! I have rebased my commits with my github username and email, force pushed the rebase, and signed the CLA. Hopefully things are OK now?

andrewgryan commented:
@twsearle thanks! That's perfect!
